policy/modules/services/spamassassin.te

```diff
@@ -384,6 +395,30 @@ tunable_policy(`spamd_enable_home_dirs',`
 	userdom_manage_user_home_content_symlinks(spamd_t)
 ')
 
+tunable_policy(`rspamd_spamd',`
+	allow spamd_t self:process setrlimit;
+	allow spamc_t self:process setrlimit;
+
+	list_dirs_pattern(spamd_t, spamd_etc_t, spamd_etc_t)
+	mmap_read_files_pattern(spamd_t, spamd_etc_t, spamd_etc_t)
+	allow spamd_t spamd_etc_t:dir watch;
+
+	mmap_read_files_pattern(spamd_t, spamd_var_lib_t, spamd_var_lib_t)
+	allow spamd_t spamd_var_lib_t:dir watch;
+	filetrans_pattern(spamd_t, spamd_var_lib_t, spamd_runtime_t, sock_file)
+
+	search_dirs_pattern(spamd_t, spamd_log_t, spamd_log_t)
+
+	fs_search_tmpfs(spamd_t)
+	manage_dirs_pattern(spamd_t, spamd_tmpfs_t, spamd_tmpfs_t)
+	manage_files_pattern(spamd_t, spamd_tmpfs_t, spamd_tmpfs_t)
+	mmap_read_files_pattern(spamd_t, spamd_tmpfs_t, spamd_tmpfs_t)
+	fs_tmpfs_filetrans(spamd_t, spamd_tmpfs_t, { dir file })
+
+	corenet_tcp_connect_http_port(spamd_t)
+	corenet_tcp_connect_redis_port(spamd_t)
+')
```

Comment:
I'm not completely convinced we need a tunable, or at least not everything needs to be in the tunable. Most of the perms  are innocuous read perms and the tmpfs ones aren't that concerning either. If the tunable is needed, then it would only need to be the network connections and maybe setrlimits IMO.

Reply:
I think it is better not to grant perms when they are not needed, hence the tunable for these rules. I don't really have a problem granting those read perms for the entire spamd domain though.